src/main/java/fi/metatavu/metaform/server/metaforms/ReplyController.java

```diff
@@ -287,6 +314,81 @@ public void deleteReply(Reply reply) {
     return anyReplyFieldDAO.listByReply(reply);
   }
 
+
+  /**
+   * Sets reply field value
+   * 
+   * @param reply reply
+   * @param name name 
+   * @param values values
+   * @return updated field
+   */
+  private ReplyField setReplyField(Reply reply, String name, List<?> values) {
+    ReplyField replyField = anyReplyFieldDAO.findByReplyAndName(reply, name);
+    if (replyField != null && !(replyField instanceof ListReplyField)) {
+      deleteField(replyField);
+      replyField = null;
+    }
+    
+    ListReplyField listReplyField = (ListReplyField) replyField;
+    if (values == null || values.isEmpty()) {
+      if (listReplyField != null) {
+        deleteListReplyFieldItems(listReplyField); 
+        listReplyFieldDAO.delete(listReplyField);
+      }
+      
+      return null;
+    }
+    
+    if (listReplyField == null) {
+      listReplyField = listReplyFieldDAO.create(UUID.randomUUID(), reply, name);
+    } else {
+      deleteListReplyFieldItems(listReplyField); 
+    }
+    
+    for (Object value : values) {
+      if (value != null) {
+        listReplyFieldItemDAO.create(UUID.randomUUID(), listReplyField, value.toString());
+      }
+    }
+    
+    return listReplyField;
+  }
+  
+  private void removeUnusedAttachmentReplyItems(List<AttachmentReplyFieldItem> items, List<String> savedFileRefs) {
+    items.stream()
+      .filter((attachmentReplyFieldItem) -> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the parentheses around the "attachmentReplyFieldItem" parameter ([squid:S1611](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS1611))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.metaform-api%3Ametaform-api&pullRequest=47&issues=AWVk3wGE4Bfu85TPTV1G&open=AWVk3wGE4Bfu85TPTV1G)

[ISSUE_KEY]: AWVk3wGE4Bfu85TPTV1G